### PR TITLE
e2e-test: stabilize tests

### LINKED ISSUE #501: e2e-test: stabilize tests
Tests to stabilize:
- `workflowitem_history_spec.js`
- `workflowitem_edit_spec.js`
- `subproject_edit_spec.js`
   - Subproject Edit When closing the project, a dialog pops up
- `subproject_permissions_spec.js`
    - Subproject Permissions Submitting the permission dialog without any changes doesn't revoke nor grant permissions and close the dialog
- `global_permissions_spec.js`
- `documents.spec`
   - Attaching a document to a workflowitem. A document can be validated.

The test list above shall be checked and stabilized. 
**Checklist:**
- Every test is independent and can be run by its own (cypress.only())
- If api calls are made in the background cypress has to wait for that route before expecting an element
- general - check the command timeouts - wait() should be way more than click() but the timeout for click() shouldn't be increased

### PR DESCRIPTION
**Pull request by daniel-arnauer**
**Checklist:**
- Every test shall be independent and can be run by its own (cypress.only())
- If api calls are made in the background cypress has to wait for that route before expecting an element

**Critical test files**:
- `workflowitem_history_spec.js`
- `workflowitem_edit_spec.js`
- `subproject_edit_spec.js`
   - Subproject Edit When closing the project, a dialog pops up
- `subproject_permissions_spec.js`
    - Subproject Permissions Submitting the permission dialog without any changes doesn't revoke nor grant permissions and close the dialog
- `global_permissions_spec.js`
- `documents.spec`
   - Attaching a document to a workflowitem. A document can be validated.

TODO:

- [x]  `workflowitem_history_spec.js`
- [x] `workflowitem_edit_spec.js`
- [x] `workflowitem_create_spec.js`
- [x] `subproject_edit_spec.js`
- [x] `subproject_search_spec.js`
- [x] `subproject_permissions_spec.js`
- [x] `global_permissions_spec.js`
- [x] `documents.spec`

Closes #501

### CODE REVIEW

**Comment by daniel-arnauer** [comment]
no problems found in subproject_permissions_spec.js

File: e2e-test/cypress/integration/documents_spec.js [hunk 1]
```diff
@@ -54,13 +58,19 @@ describe("Attaching a document to a workflowitem.", function() {
     const fileDisplayName = "Validation_Test";
     uploadDocument(fileDisplayName, fileName);
     // submit and close the dialog:
-    cy.get("[data-test=submit]").click();
+    cy.get("[data-test=submit]")
+      .should("be.visible")
+      .click();
```
Comment (Stezido) [inline e2e-test/cypress/integration/documents_spec.js:64]: wait for subproject.viewdetails before opening the info dialog. If it is too fast the document isn't shown in the documents tab.

File: e2e-test/cypress/integration/documents_spec.js [hunk 2]
```diff
@@ -71,33 +81,46 @@ describe("Attaching a document to a workflowitem.", function() {
...
   });
 
   it("Validation of wrong document fails.", function() {
     const fileDisplayName = "Wrong_Document_Test";
     uploadDocument(fileDisplayName, fileName);
     // submit and close the dialog:
-    cy.get("[data-test=submit]").click();
+    cy.get("[data-test=submit]")
+      .should("be.visible")
+      .click();
```
Comment (Stezido) [inline e2e-test/cypress/integration/documents_spec.js:97]: wait for subproject.viewdetails before opening the info dialog. If it is too fast the document isn't shown in the documents tab.

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,7 +15,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 <!-- ### Removed -->
 
-<!-- ### Fixed -->
+### Fixed
+
+- Stabilize some e2e-tests [#501](https://github.com/openkfw/TruBudget/issues/501)
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply CHANGELOG.md]: changelog removed

File: e2e-test/cypress/integration/documents_spec.js (hunk 1 above)
Comment (daniel-arnauer) [reply e2e-test/cypress/integration/documents_spec.js:71]: route and wait is now set

File: e2e-test/cypress/integration/documents_spec.js (hunk 2 above)
Comment (daniel-arnauer) [reply e2e-test/cypress/integration/documents_spec.js:108]: route and wait is now set

**Pull request merged**